pkg/util/utils.go

```diff
@@ -53,13 +54,18 @@ type RetriableError struct {
 
 func (r RetriableError) Error() string { return "Temporary Error: " + r.Err.Error() }
 
-// CalculateDiskSizeInMB returns the number of MB in the human readable string
-func CalculateDiskSizeInMB(humanReadableDiskSize string) int {
-	diskSize, err := units.FromHumanSize(humanReadableDiskSize)
+// CalculateSizeInMB returns the number of MB in the human readable string
+func CalculateSizeInMB(humanReadableSize string) int {
+	_, err := strconv.ParseInt(humanReadableSize, 10, 64)
+	if err == nil {
+		humanReadableSize += "mb"
+	}
+	size, err := units.FromHumanSize(humanReadableSize)
 	if err != nil {
-		glog.Errorf("Invalid disk size: %v", err)
+		exit.WithCode(exit.Config, "Invalid size passed in argument: %v", err)
```

Comment:
I agree with you that the most ideal way would be to let the caller handle it. But, before I pushed this commit, I tried to do that and it became pretty messy. Adding loads of `if` conditions didn't make sense when ideally we should be exiting immediately if we are not able to parse the size. It seems to be better for the function itself to exit rather than the chance of caller not handling it properly deep in the code.

This utility function needs to be called every time when we need to parse any memory or disk size. Be it the defaults or input by the user. Do you think it is still adding a lot of `if` conditions to check for errors everywhere?

Reply:
ok you convinced me this is a case we can tolerate exit because of the nature that if it is not readable we can not for sure continue ! we could be more cautious if more people use this function for other utilities.